### PR TITLE
Fix issue #159: Failure when cross-compiling for VxWorks

### PR DESCRIPTION
**Pull request by moberhuber**
When cross-compiling, the cmake/extensions/CMakeLists.txt may be loaded without ${MSVC_VERSION} set. Protect against errors by making the code conditional.

### CODE REVIEW

File: cmake/extensions/CMakeLists.txt [hunk 1]
```diff
@@ -1,3 +1,4 @@
+if(${MSVC_VERSION})
```
Comment (thewtex) [inline cmake/extensions/CMakeLists.txt]: Per CMake style, we can use `if(MSVC_VERSION)` here.

**Comment by moberhuber** [comment]
Thanks for the feedback, I've cleaned up as requested (wanted a minimal source diff initially).
Please review again, thanks !

**Comment by moberhuber** [comment]
I've added 2 more fixes for cross-compiling VxWorks, could these be considered please? - Thx :)

**Comment by thewtex** [comment]
@moberhuber thanks!

LGTM

CC @jcfr

**Comment by jcfr** [comment]
Fixes related to libutil have been pushed into their own PR. See https://github.com/python-cmake-buildsystem/python-cmake-buildsystem/pull/165

**Comment by jcfr** [comment]
Fix related to `sys/select.h` has been pushed into its own PR. See https://github.com/python-cmake-buildsystem/python-cmake-buildsystem/pull/166

**Comment by jcfr** [comment]
Now for de1ffdf and fdd2c56, the exclusion of MSVC is already done when the version is empty or doesn't match, the issue happen because `CMAKE_SIZEOF_VOID_P` is not set.

With a working toolchain, I would expect this variable to be set to either `4` (32-bit) or `8` (64-bit), I suspect something is missing from the toolchain and/or companion files.

For reference this variable is set within CMake [here](https://github.com/Kitware/CMake/blob/60d80bca4afcfb3a38b588a5a2060cc275b4afbc/Modules/CMakeCCompiler.cmake.in#L45-L47) using the content of the variable `CMAKE_C_SIZEOF_DATA_PTR` originally set here: https://github.com/Kitware/CMake/blob/60d80bca4afcfb3a38b588a5a2060cc275b4afbc/Modules/CMakeDetermineCompilerABI.cmake#L41-L59

Looking at the `CMakeOutput.log` should help understand better.

@moberhuber Let us know what you find out.

**Comment by moberhuber** [comment]
Hi @jcfr , regarding fdd2c56 I can confirm that the issue does not occur with the GNU toolchain which does set `CMAKE_SIZEOF_VOID_P` as part of its compiler detection.

But for VxWorks, we also want to build with the Diab toolchain which isn't supported by stock CMake so far. So we use our own (very basic) cross-toolchain information. This works fine everywhere else, in my fork I could build a basic Python that works even without `CMAKE_SIZEOF_VOID_P` being set (I suspect because any important compiler / platform information is discovered directly through other mechansims).

**Comment by jcfr** [comment]
May  be you could add the following to the toolchain file:

```
set(CMAKE_SIZEOF_VOID_P 8)
```

Or we could add a statement like this in the python cmake build system:

```
if(NOT DEFINED CMAKE_SIZEOF_VOID_P AND CMAKE_CROSSCOMPILING)
  set(CMAKE_SIZEOF_VOID_P 8)
  message(STATUS "Defaulting CMAKE_SIZEOF_VOID_P to 8 because CMakeDetermineCompilerABI failed to do so")
endif()
```

Let me know what you think.

Jc

**Comment by jcfr** [comment]
To summarize:
* Fixes related to `libutil` have been pushed into their own PR. See #165
* Fix related to `sys/select.h` has been pushed into its own PR. See #166


If the last issue related to `CMAKE_SIZEOF_VOID_P` is still relevant, consider creating a new issue and evaluating the approach proposed in https://github.com/python-cmake-buildsystem/python-cmake-buildsystem/pull/160#issuecomment-261821592

**Pull request closed without merging**